File: app/src/main/java/com/maxdexter/mynote/database/NoteDao.java
package com.maxdexter.mynote.database;



import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.maxdexter.mynote.data.Note;

import java.util.List;


@Dao
public interface NoteDao {
    @Query("SELECT * FROM note")
    List<Note> getAll();

    @Query("SELECT * FROM note WHERE mUUID = :uuid")
    Note getById(String uuid);

    @Insert(onConflict =  OnConflictStrategy.REPLACE)
    void insert(Note note);

    @Update
    void update(Note note);

    @Delete
    void delete(Note note);
}

File: app/src/main/java/com/maxdexter/mynote/database/NoteDao.kt
package com.maxdexter.mynote.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.maxdexter.mynote.model.Note

@Dao
interface NoteDao {
    @Query("SELECT * FROM note")
    fun getAll():LiveData<List<Note?>>

    @Query("SELECT * FROM note WHERE uuid = :uuid")
    fun getById(uuid: String): LiveData<Note>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(note: Note)

    @Update
    fun update(note: Note)

    @Delete
    fun delete(note: Note)
}